subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
So given Go 1.6 we can:

1. Fail when user wants to create sse subscription with client that has client.Timeout set
2. Add another http.client to Config to allow user to specify custom http.Client (config.HTTPSSEClient) that is used for sse subscriptions. This can be set to config.HTTPClient when not set. First point will also apply but then he can use config.HTTPClient with Timeout set